src/SumoCoders/FrameworkCoreBundle/BreadCrumb/BreadCrumbBuilder.php

```diff
@@ -0,0 +1,248 @@
+                    $breadCrumb->setUri($temporaryItem->getUri());
+                }
+                $items[] = $breadCrumb;
+
+                $temporaryItem = $temporaryItem->getParent();
+            }
+
+            $home = new MenuItem('core.menu.home', $this->factory);
+            $home->setLabel('core.menu.home');
+            $home->setUri('/' . $locale);
+            $this->items[] = $home;
+
+            $this->items = array_merge($this->items, array_reverse($items));
+        }
+    }
+
+    /**
+     * Grab the menu
+     *
+     * This method will use the ConfigureMenuEvent to get all the items
+     *
+     * @return \Knp\Menu\ItemInterface
+     */
+    protected function getTheMenu()
+    {
+        $menu = $this->factory->createItem('root');
+
+        $this->eventDispatcher->dispatch(
+            ConfigureMenuEvent::EVENT_NAME,
+            new ConfigureMenuEvent(
+                $this->factory,
+                $menu
+            )
+        );
+
+        return $menu;
+    }
+
+    /**
+     * Find an item in the menu based on its URI
+     *
+     * @param MenuItem $menuItem
+     * @param          $uri
+     * @return MenuItem|null
+     */
+    protected function findItemBasedOnUri(MenuItem $menuItem, $uri)
+    {
+        if ($uri === $menuItem->getUri()) {
+            return $menuItem;
+        }
+
+        if (!$menuItem->hasChildren()) {
+            return null;
+        }
+
+        foreach ($menuItem->getChildren() as $child) {
+            $item = $this->findItemBasedOnUri(
+                $child,
+                $uri
+            );
+
+            if (null !== $item) {
+                return $item;
+            }
+        }
+
+        return null;
+    }
+
+    /**
+     * Extract the items for the breadcrumb based on request
+     *
+     * @param Request $request
+     */
+    protected function extractItemsBasedOnTheRequest(Request $request)
+    {
+        if ($this->dontExtractFromTheRoute) {
+            return;
+        }
+
+        $this->extractItemsBasedOnUri(
+            $request->getPathInfo(),
+            $request->getLocale()
+        );
+    }
+
+    /**
+     * @param Request $request
+     * @return \Knp\Menu\ItemInterface
+     */
+    public function createBreadCrumb(Request $request)
+    {
+        $this->extractItemsBasedOnTheRequest($request);
+
+        $menu = $this->factory->createItem('root');
+        $menu->setChildrenAttribute('class', 'breadcrumb');
+
+        foreach ($this->items as $i => $item) {
+            // the last item shouldn't have a link
+            if ((count($this->items) - 1) == $i) {
+                $item->setUri(null);
+            }
+
+            $menu->addChild($item);
+        }
+
+        return $menu;
+    }
+
+    /**
+     * @param bool $isCustom
+     * @return BreadCrumbBuilder
+     */
+    public function setDontExtractFromTheRequest($isCustom = true)
```

Comment:
als ge je variabele hernoemt kan je hier dan dus `$this->extractFromTheRoute = false;` van maken